include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -4665,6 +4663,94 @@ FEEvaluationBase<dim, n_components_, Number, is_face, VectorizedArrayType>::
   for (unsigned int v = n_filled_lanes; v < VectorizedArrayType::size(); ++v)
     dof_indices[v] = numbers::invalid_unsigned_int;
 
+  if (vectors_sm[0] != nullptr)
+    {
+      const auto compute_vector_ptrs = [&](const unsigned int comp) {
+        std::array<typename VectorType::value_type *,
+                   VectorizedArrayType::size()>
+          vector_ptrs = {};
+
+        /*
+        for (unsigned int v = 0; v < n_filled_lanes; ++v)
+          {
+            const auto &temp =
+              this->dof_info->dof_indices_contiguous_sm
+                [ind][this->cell * VectorizedArrayType::size() + v];
+
+            if (temp.first != numbers::invalid_unsigned_int)
+              vector_ptrs[v] = const_cast<typename VectorType::value_type *>(
+                vectors_sm[comp]->operator[](temp.first).data() + temp.second);
+            else
+              vector_ptrs[v] = nullptr;
+          }
+        for (unsigned int v = n_filled_lanes; v < VectorizedArrayType::size();
+             ++v)
+          vector_ptrs[v] = nullptr;
```

Comment:
@kronbichler I have slightly restructured the code. I think this is how `DoFInfo::dof_indices_contiguous_sm` might eventually be used.